test/kubernetes_unittest.cc

```diff
@@ -72,6 +81,11 @@ class KubernetesTest : public ::testing::Test {
       throw(json::Exception) {
     return reader->UpdateServiceToPodsCache(endpoints, is_deleted);
   }
+
+  static void SetServiceAccountDirectoryForTest(
```

Comment:
Ah, NVM. I figured it out. It's because of [this line](https://github.com/Stackdriver/metadata-agent/blob/master/src/kubernetes.h#L69):

```  friend class KubernetesTest;```